src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -244,6 +246,233 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
             }
         }
 
+        internal static ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> GetAnalyzerConfigOptions(
+            IReadOnlyList<string> sourcePaths,
+            ArrayBuilder<EditorConfig> analyzerConfigs,
+            CommonMessageProvider messageProvider,
+            DiagnosticBag diagnostics)
+        {
+            // Sort editorconfig paths from shortest to longest.
+            // Since paths are compared as ordinal string comparisons, the least nested
+            // file will always be the first in sort order
+            analyzerConfigs.Sort(Comparer<EditorConfig>.Create(
+                (e1, e2) => e1.Directory.Length.CompareTo(e2.Directory.Length)));
+
+            var allDiagnosticOptions =
+                ArrayBuilder<ImmutableDictionary<string, ReportDiagnostic>>.GetInstance(sourcePaths.Count);
```

Comment:
Even if the intent is ordinal comparison consider using an explicit comparer here. This makes it clear to later developers that the scenario was considered.

Reply:
I think you're thinking of the ImmutableDictionary. This is an ImmutableArray of ImmutableDictionaries, so the only comparer I expect to work here is reference comparison. The later creation of the ImmutableDictionaries should have explicit comparers.